src/main/kotlin/org/jitsi/rtp/ByteArrayBuffer.java

```diff
@@ -87,6 +89,58 @@ public void readRegionToBuff(int off, int len, byte[] outBuff)
         System.arraycopy(this.buffer, startOffset, outBuff, 0, len);
     }
 
+    /**
+     * Grows the internal buffer of this {@code ByteArrayBuffer}.
+     *
+     * This will change the data buffer of this packet but not the length of the
+     * valid data. Use this to grow the internal buffer to avoid buffer
+     * re-allocations when appending data.
+     *
+     * @param howMuch the number of bytes by which this {@code NewRawPacket} is to
+     * grow
+     */
+    public void grow(int howMuch) {
+        if (howMuch < 0)
+            throw new IllegalArgumentException("howMuch: " + howMuch);
+
+        int newLength = length + howMuch;
+
+        if (newLength > buffer.length - offset) {
+            byte[] newBuffer = BufferPool.Companion.getGetArray().invoke(newLength);
+
+            System.arraycopy(buffer, offset, newBuffer, 0, length);
+            offset = 0;
+            byte[] oldBuffer = buffer;
+            setBuffer(newBuffer);
+            BufferPool.Companion.getReturnArray().invoke(oldBuffer);
+        }
+    }
+
+    /**
+     * Shrink the buffer of this packet by specified length
+     *
+     * @param len length to shrink
+     */
+    public void shrink(int len)
+    {
+        if (len <= 0)
+            return;
+
+        this.length -= len;
+        if (this.length < 0)
+            this.length = 0;
+    }
+
+
+
+    /**
+     * @param buffer the buffer to set
+     */
+    public void setBuffer(byte[] buffer)
```

Comment:
Can we avoid exposing this?

Reply:
i think we probably can, i'll take a look at doing it in a future change